clang/lib/Sema/SemaBounds.cpp

```diff
@@ -5543,6 +5523,22 @@ void Sema::WarnDynamicCheckAlwaysFails(const Expr *Condition) {
   }
 }
 
+// If the VarDecl D has a byte_count or count bounds expression,
+// NormalizeBounds expands it to a range bounds expression.  The expanded
+// range bounds are attached to the VarDecl D to avoid recomputing the
+// normalized bounds for D.
+BoundsExpr *Sema::NormalizeBounds(const VarDecl *D) {
+  // If D already has a normalized bounds expression, do not recompute it.
+  if (const BoundsExpr *NormalizedBounds = D->getNormalizedBounds()) 
```

Comment:
`setNormalizedBounds` already set non-const `BoundsExpr`. So I guess you do not need the const_cast here.

Reply:
Since the `VarDecl D` is `const`, `D->getNormalizedBounds()` seems to return a `const BoundsExpr`. I could put the `const_cast` on `D` instead.

Reply:
As discussed offline, I removed the `const` from `VarDecl::getNormalizedBounds`.